### ISSUE TITLE
GeoTiff writer is distorting data

### ISSUE THREAD

**Issue by metasim** [locationtech/rasterframes#224]
I read and wrote to gtiff a single MODIS granule (single channel). In the resulting `gdalinfo` the pixel size is (460.720950959172399,-460.720950959172399) 

But the original granule's pixel size is slightly larger: (463.312716527916905,-463.312716527916677)

See the movie attached, you can kind of see an artifact that may be the tile boundaries.

Happy to share original and written file.

[ScreenRecordModis.mov.zip](https://github.com/locationtech/rasterframes/files/3448602/ScreenRecordModis.mov.zip)

The tile dimensions are pretty much normal for this kind of file, so pretty irregular. The most common one is not even square.

```
+------------------+-----+
|rf_dimensions(B01)|count|
+------------------+-----+
|         [256, 96]|    6|
|        [257, 257]|   18|
|         [257, 96]|    3|
|          [97, 96]|    1|
|         [97, 257]|    6|
|        [256, 257]|   36|
|        [257, 256]|    9|
|        [256, 256]|   18|
|         [97, 256]|    3|
+------------------+-----+
```

_Originally posted by @vpipkt in https://github.com/locationtech/rasterframes/pull/212#issuecomment-516569592_

**Comment by vpipkt** [comment]
So in the example I posted above, or a very similar replication the overall corner coordinates are off by about 1/2 the pixel size. Our upper left corner matches input file but our lower right is about 1/2 pixel further east and south. 

The interesting part is we have about 1% more pixels in our written product. 

The bit of border containing the half pixel on the south and east edges of our written product is full of nodata. 

Still investigating.

**Comment by vpipkt** [comment]
I don't htink this is affecting this particular test case, and is probably pretty rare but this should probably take individual min of cell height and width individually as opposed to smaller `resolution`. 

But not 100% sure about that either.

https://github.com/locationtech/rasterframes/blob/develop/core/src/main/scala/org/locationtech/rasterframes/expressions/aggregates/ProjectedLayerMetadataAggregate.scala#L142


Also i assume lots of `resolution` comparisons [could be expensive  ](https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/CellSize.scala#L31)

**Comment by vpipkt** [comment]
I am looking at the use of GeoTrellis [`LayoutScheme`](https://geotrellis.github.io/scaladocs/latest/#geotrellis.spark.tiling.LayoutScheme) ...

> ... resolution... will not necessarily match the CellSize provided...

Investigating work arounds

**Comment by vpipkt** [comment]
Possible fix in the works in pr #231

**Comment by metasim** [comment]
@vpipkt 

> So in the example I posted above, or a very similar replication the overall corner coordinates are off by about 1/2 the pixel size. Our upper left corner matches input file but our lower right is about 1/2 pixel further east and south.

This makes me wonder if there's an improper use of the `clamp` parameter in `geotrellis.raster.RasterExtent.gridBoundsFor`...

**Comment by metasim** [comment]
Like here:

https://github.com/locationtech/rasterframes/blob/86b64ed1d6f4b5634cbe0cc6b8418b50021bb1c0/core/src/main/scala/org/locationtech/rasterframes/expressions/aggregates/TileRasterizerAggregate.scala#L100

**Comment by vpipkt** [comment]
Hopefully with improvement of #231 and work to fix #242, we should be able to call this fixed. If we can't fix before 0.8.0 release, we ought to note it as a known bug.
